File: backend/Elearning.API/Models/CourseCategory.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Elearning.API.Models;

[Table("CourseCategory")]
public partial class CourseCategory
{
    [Key]
    public int CourseCategoryId { get; set; }

    [StringLength(200)]
    public string Name { get; set; } = null!;

    [StringLength(1000)]
    public string? Description { get; set; }

    public bool IsActive { get; set; }

    [InverseProperty("CourseCategory")]
    public virtual ICollection<Course> Courses { get; set; } = new List<Course>();
}

File: frontend/elearning-web/src/types/courseCategory.ts
export type CourseCategoryDto = {
  id: number;
  name: string;
  description?: string | null;
  isActive: boolean;
};

export type CourseCategoryCreateDto = {
  name?: string | null;
  description?: string | null;
};

export type CourseCategoryEditDto = {
  id: number;
  name?: string | null;
  description?: string | null;
};
